internal/repositories/dcn.go

```diff
@@ -14,14 +17,44 @@ func DCNToAPI(d *models.DCN) *gmodel.Dcn {
 		Node:      d.Node,
 		ExpiresAt: d.Expiration.Ptr(),
 		Name:      d.Name.Ptr(),
+		VehicleID: d.VehicleID.Ptr(),
 	}
 }
 
-func (r *Repository) GetDCNByNode(ctx context.Context, node []byte) (*gmodel.Dcn, error) {
-	dcn, err := models.DCNS(models.DCNWhere.Node.EQ(node)).One(ctx, r.pdb.DBS().Reader)
+func (r *Repository) getDCNByNode(ctx context.Context, dcnParams model.DCNBy) (*gmodel.Dcn, error) {
+	dcn, err := models.DCNS(
+		models.DCNWhere.Node.EQ(dcnParams.Node),
+	).One(ctx, r.pdb.DBS().Reader)
 	if err != nil {
 		return nil, err
 	}
 
 	return DCNToAPI(dcn), nil
 }
+
+func (r *Repository) getDCNByName(ctx context.Context, dcnParams model.DCNBy) (*gmodel.Dcn, error) {
+	dcn, err := models.DCNS(
+		models.DCNWhere.Name.EQ(null.StringFrom(*dcnParams.Name)),
+	).One(ctx, r.pdb.DBS().Reader)
+	if err != nil {
+		return nil, err
+	}
+
+	return DCNToAPI(dcn), nil
+}
+
+func (r *Repository) GetDCN(ctx context.Context, params model.DCNBy) (*gmodel.Dcn, error) {
```

Comment:
The repository seems like a strange place for this. Could we do it in the resolver?

Reply:
I view the resolver more like a Router in the MVC architecture. Moving this logic there would mean its involved in validation, which is a logic i don't think belongs in a router, since its basically meant to just make sure the right handler gets called.

Reply:
You're putting it in the repository, the storage layer. If you want to introduce another layer, that's fine.

In the interest of getting this moving—if the two functions this calls had narrower input types, I'd be happier.

Reply:
So, it has to go one of two bad places. I have a preference, you have a preference, that's fine; what I don't like is `GetByName(BigUnionType)`.